src/pages/storage/panels/EditStorageBucketPanel.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  ActionButton,
+  Button,
+  useNotify,
+  useToastNotification,
+} from "@canonical/react-components";
+import SidePanel from "components/SidePanel";
+import type { FC } from "react";
+import { useState } from "react";
+import usePanelParams from "util/usePanelParams";
+import * as Yup from "yup";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import ScrollableContainer from "components/ScrollableContainer";
+import ResourceLink from "components/ResourceLink";
+import StorageBucketForm from "../forms/StorageBucketForm";
+import type { StorageBucketFormValues } from "../forms/StorageBucketForm";
+import { updateStorageBucket } from "api/storage-buckets";
+import { queryKeys } from "util/queryKeys";
+import { useQueryClient } from "@tanstack/react-query";
+import { testDuplicateStorageBucketName } from "util/storageBucket";
+import type { LxdStorageBucket } from "types/storage";
+import { pluralize } from "util/instanceBulkActions";
+
+interface Props {
+  bucket: LxdStorageBucket;
+}
+const EditStorageBucketPanel: FC<Props> = ({ bucket }) => {
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const controllerState = useState<AbortController | null>(null);
+  const queryClient = useQueryClient();
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const bucketSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(
+        ...testDuplicateStorageBucketName(
+          panelParams.project,
+          controllerState,
+          panelParams.bucket ?? "",
+        ),
+      )
+      .required("Bucket name is required"),
+  });
+
+  const handleSuccess = (bucketName: string) => {
+    toastNotify.success(
+      <>
+        Bucket{" "}
+        <ResourceLink
+          type="bucket"
+          value={bucketName}
+          to={`/ui/project/${panelParams.project}/storage/buckets`}
+        />{" "}
+        updated.
+      </>,
+    );
+    closePanel();
+  };
+
+  const formik = useFormik<StorageBucketFormValues>({
+    initialValues: {
+      name: bucket?.name,
+      pool: bucket?.pool,
+      size: bucket?.config.size,
+      description: bucket?.description,
+    },
```

Comment:
We should preserve the target in the values as well, so we can use it on submit.